cmake/SetupCxxFlags.cmake

```diff
@@ -29,10 +36,18 @@ if(NOT "${OVERRIDE_ARCH}" STREQUAL "OFF")
       # a fix that allows this vectorization flag again.
       $<$<COMPILE_LANGUAGE:CXX>:-march=${OVERRIDE_ARCH} -mno-avx512f>)
 else()
-  set_property(TARGET SpectreFlags
-      APPEND PROPERTY
-      INTERFACE_COMPILE_OPTIONS
-      $<$<COMPILE_LANGUAGE:CXX>:-march=native -mno-avx512f>)
+  # Apple Silicon Macs do not support the -march flag or the -mno-avx512f flag
+  if(APPLE AND "${CMAKE_HOST_SYSTEM_PROCESSOR}" STREQUAL "arm64")
+    set_property(TARGET SpectreFlags
+        APPEND PROPERTY
+        INTERFACE_COMPILE_OPTIONS
+        $<$<COMPILE_LANGUAGE:CXX>:>)
```

Comment:
I believe this command doesn't do anything, so better delete this clause entirely?

Reply:
Changed to read
~~~
if(NOT APPLE OR NOT "${CMAKE_HOST_SYSTEM_PROCESSOR}" STREQUAL "arm64")
    set_property(TARGET SpectreFlags
        APPEND PROPERTY
        INTERFACE_COMPILE_OPTIONS
        $<$<COMPILE_LANGUAGE:CXX>:-march=native -mno-avx512f>)
  endif()
~~~